### DISCUSSION TITLE
Implementing swim lanes in free version of react-flow

### DISCUSSION THREAD

**Discussion by ScottEL1** [xyflow/xyflow#2359] [Layouting]
My team wants to implement swim lanes in our GUI.  Swim lanes could be vertical or horizontal.  Lanes could represent layers of an organization or a network, etc.  Lanes contain nodes that may be linked to each other and linked to nodes outside of their lanes.  Lanes need to be collapsible so that users can compare nodes in lane A to nodes in lane X.  Has anyone done this or can point us to an example?

**Comment by moklick** [comment]
It's not built-in but you can create swim lanes with custom nodes. Stripe did something similar with a custom node type: https://stripe.com/docs/payments/checkout/how-checkout-works#lifecycle

**Reply by ScottEL1** [reply]
Thanks!  I will point the devs to your example.

**Reply by ZakKa89** [reply]
I would love to be able to see their implementation :D

**Reply by ZakKa89** [reply]
It doesn't look like a custom node type will be enough, since there can be multiple nodes within one swimming lane. 

Do you think using a subflow as a swimming lane (maybe customize it if possible) would work @moklick ? https://reactflow.dev/docs/guides/sub-flows/

**Comment by sunillarsson** [comment]
Here's an implementation I found while looking for a swimlane layout as well:

https://github.com/liang-faan/reactflow-swimlane
